src/ruby/ext/grpc/rb_server.c

```diff
@@ -326,7 +327,14 @@ static VALUE grpc_rb_server_add_http2_port(VALUE self, VALUE port,
                StringValueCStr(port));
     }
   } else {
-    creds = grpc_rb_get_wrapped_server_credentials(rb_creds);
+    if (grpc_rb_is_server_credentials(rb_creds)) {
```

Comment:
nit: can we add a TODO comment here that if we create a common parent class for all server-side credentials, then we can have a single method to retrieve the underlying `grpc_server_credentials` object, and avoid the need for this reflection ?

Reply:
done